src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -0,0 +1,245 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+from __future__ import annotations
+
+from typing import Any, Callable, Hashable, Iterator, Literal
+
+import numpy as np
+import pandas as native_pd
+from pandas._typing import ArrayLike, DtypeObj, NaPosition, Self
+from pandas.core.arrays import ExtensionArray
+from pandas.core.dtypes.base import ExtensionDtype
+from pandas.core.indexes.frozen import FrozenList
+
+
+# modin index class, wrapper to native pandas for now
+class Index:
```

Comment:
Not sure if this was considered, but rather than having a class with a bunch of methods that are the same as the pandas.Index, have we considered doing something like this inheriting from the pandas Index class and just overwriting any calls to the method attributes to do something like super().method(self.to_pandas(), *args, **kwargs)? That way, we get support for all of the methods (including the ones we don't need distributed e.g. set_names), and can go back and add support for distributed methods.

Reply:
please post result of discussion here

Reply:
Not sure if a final decision has been made, added you to the thread